hera_mc/mc_session.py

```diff
@@ -1396,6 +1398,306 @@ def get_node_power_command(self, most_recent=None, starttime=None, stoptime=None
                                  filter_column='node', filter_value=nodeID)
 
+    def add_correlator_config_file(self, config_hash, config_file):
+        """
+        Add new correlator config file to the M&C database.
+
+        Parameters:
+        ------------
+        config_hash: string
+            unique hash of the config
+        config_file: string
+            name of the config file in the Librarian
+        """
+        from .correlator import CorrelatorConfigFile
+
+        self.add(CorrelatorConfigFile.create(config_hash, config_file))
+
+    def get_correlator_config_file(self, config_hash):
+        """
+        Get a correlator config file record from the M&C database.
+
+        Parameters:
+        ------------
+        config_hash: string
+            unique hash for config file
+
+        Returns:
+        --------
+        list of CorrelatorConfigFile objects
+        """
+        from .correlator import CorrelatorConfigFile
+
+        query = self.query(CorrelatorConfigFile).filter(CorrelatorConfigFile.config_hash == config_hash)
+
+        return query.all()
+
+    def add_correlator_config_status(self, time, config_hash):
+        """
+        Add new correlator config status to the M&C database.
+
+        Parameters:
+        ------------
+        time: astropy time object
+            astropy time object based on a timestamp reported by the correlator
+        config_hash: string
+            unique hash of the config
+        """
+        from .correlator import CorrelatorConfigStatus
+
+        self.add(CorrelatorConfigStatus.create(time, config_hash))
+
+    def get_correlator_config_status(self, most_recent=None, starttime=None,
+                                     config_hash=None, stoptime=None, state_type=None):
+        """
+        Get correlator config status record(s) from the M&C database.
+
+        Parameters:
+        ------------
+        most_recent: boolean
+            if True, get most recent record. Defaults to True if starttime is None.
+
+        starttime: astropy time object
+            Time to look for records after. Ignored if most_recent is True,
+            required if most_recent is False.
+
+        stoptime: astropy time object
+            Last time to get records for, only used if starttime is not None.
+            If none, only the first record after starttime will be returned.
+            Ignored if most_recent is True.
+
+        config_hash: string
+            unique hash for config file
+
+        Returns:
+        --------
+        list of CorrelatorConfigStatus objects
+        """
+        from .correlator import CorrelatorConfigStatus
+
+        return self._time_filter(CorrelatorConfigStatus, 'time', most_recent=most_recent,
+                                 starttime=starttime, stoptime=stoptime,
+                                 filter_column='config_hash', filter_value=config_hash)
+
+    def _add_config_file_to_librarian(self, config, config_hash,
```

Comment:
@pkgw can you take a look at this method? It's for adding config files to the librarian and I want to make sure it's right.

Reply:
Paul thinks it is.